nigsp/tests/test_metrics.py

```diff
@@ -72,6 +72,25 @@ def test_gsdi():
     assert (gsdi_out["beta_over_alpha"] == gsdi_in).all()
 
 
+def test_smoothness():
+    s1 = rand(10)
+    s2 = rand(10, 2)
+    s3 = rand(2, 10)
+    laplacian = rand(10, 10)
+
+    expected_smoothness1 = np.dot(s1.T, np.dot(laplacian, s1))
```

Comment:
Is this correct? If so you don't need a warning and a special treatment, since it's the same as the main case!

Reply:
I'm talking about s1-s2

Reply:
ah yes that's correct sorry, I'm trying to make it work with a 3rd dimension for different subjects so I'll see how this evolves as I will probably move from np.matmul to np.tensordot

Reply:
Not sure it'd fit in in memory, keep an eye! You might have to run over subjects dim with a loop

Reply:
Tensordot might indeed not work out - Maybe you can constrain axes in the multiplication, or indeed loop through subjects. I have some code in another function (the graph fourier transform, if I remember correctly) that split cases based on dimensions (and runs a loop).